apps/aecore/lib/aecore/channel/channel_state_peer.ex

```diff
@@ -0,0 +1,608 @@
+defmodule Aecore.Channel.ChannelStatePeer do
```

Comment:
the naming here uses `fsm`, but as I understand it is not a real finite state machine, right?

Reply:
Yes, the `highest_state` and `highest_signed_state` are part of the state. I call the **type** of state `fsm_state` to distinguish from `highest_state` and `highest_signed_state`. 

It's a real final state machine if you ignore `highest_state`, `highest_signed_state` and function arguments. When you call a function `fsm_state` changes in a predictable way. So it has an FSM in it's core, but there are other properties.

Reply:
In epoch there is the need for some inbetween - waiting - states, this is why I ask, but lets keep it like this.

Reply:
I want to split the negotiation and data exchange part of channel with security related handling of the channel state. (I think that's still mixed in epoch.) That's why there is no timeout code and most of waiting states in my implementation. With my code if you don't fully understand channels you can still write a secure exchange of channel data and channel negotiation. (of course to a reasonable degree)